packages/shared_preferences/shared_preferences_ios/lib/shared_preferences_ios.dart

```diff
@@ -2,52 +2,71 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:async';
-
 import 'package:flutter/services.dart';
 import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+import 'messages.g.dart';
+
+typedef _Setter = Future<void> Function(String key, Object value);
 
-const MethodChannel _kChannel =
-    MethodChannel('plugins.flutter.io/shared_preferences_ios');
+/// iOS implementation of shared_preferences.
+class SharedPreferencesIos extends SharedPreferencesStorePlatform {
+  final UserDefaultsApi _api = UserDefaultsApi();
+  late final Map<String, _Setter> _setters = <String, _Setter>{
+    'Bool': (String key, Object value) {
+      return _api.setBool(key, value as bool);
+    },
+    'Double': (String key, Object value) {
+      return _api.setDouble(key, value as double);
+    },
+    'Int': (String key, Object value) {
+      return _api.setValue(key, value as int);
+    },
+    'String': (String key, Object value) {
+      return _api.setValue(key, value as String);
+    },
+    'StringList': (String key, Object value) {
+      return _api.setValue(key, value as List<String?>);
+    },
+  };
 
-/// The macOS implementation of [SharedPreferencesStorePlatform].
-///
-/// This class implements the `package:shared_preferences` functionality for iOS.
-class SharedPreferencesIOS extends SharedPreferencesStorePlatform {
```

Comment:
Renaming the class is a breaking change.

It's also still not actually clear to me what the correct capitalization is. The [relevant Dart rule](https://dart.dev/guides/language/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words) says that two-letter acronyms are capitalized in Dart. "i" isn't part of the acronym; the acronym itself, OS, is two letters. So there are two possible interpretations:
- It's a single letter ("i"), which is capitalized because it's the first letter, followed by a two-letter acronym, which is also capitalized, giving `IOS`.
- We pretend it's a normal three-letter word (or a three-letter acronym-like thing?), giving `Ios`.

Dart appears to use the former ([ref](https://api.dart.dev/stable/2.16.1/dart-io/Platform/isIOS.html)), and I only just realized that that there's an explicit mention in the Flutter style guide saying it should be the latter (although that section is itself somewhat wrong since it says Dart rules would give `macOs`, which isn't correct). Given the existing inconsistency in the ecosystem, I haven't caught one vs. the other in reviews.

I'll make sure to apply the Flutter rule going forward now that I'm aware of it, but it's not worth a breaking change here to fix.

Reply:
Sure, done.